### PR TITLE
[CELEBORN-644][HELM] Support Helm Deploy Celeborn with HostNetwork And DnsPolicy

### PR DESCRIPTION
**Pull request by zwangsheng**
### What changes were proposed in this pull request?
Support service type `NodePort` in `values.yaml`

### Why are the changes needed?
For Kubernetes IT, should start celeborn cluster in minikube with nodePort for discover

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
CI Test & Local Test

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Wait, exposing NodePort at Service granularity does not make sense, we need to access the Pod directly.

**Comment by zwangsheng** [comment]
> Wait, exposing NodePort at Service granularity does not make sense, we need to access the Pod directly.

Looks like `hostNetwork` better suited to our needs.

**Comment by pan3793** [comment]
The host network is discouraged in K8s workload.

Technically, the Pod network should be reachable from the GA runner, better to investigate why it is not first.

**Comment by pan3793** [comment]
let's try https://minikube.sigs.k8s.io/docs/drivers/none/

**Comment by zwangsheng** [comment]
Given Kubernetes' diverse network setup, the comparison depends on the user's infrastructure. 

Therefore, we still choose simple `hostNetwork` to realize the access inside kubernetes cluster  from outside kubernetes cluster.

**Comment by zwangsheng** [comment]
As for IT mirror test, 

In github runner, you can reach `minikube ip`

```
# minikube ip | xargs ping -c 10
64 bytes from 192.168.49.2: icmp_seq=1 ttl=64 time=0.054 ms
64 bytes from 192.168.49.2: icmp_seq=2 ttl=64 time=0.048 ms
64 bytes from 192.168.49.2: icmp_seq=3 ttl=64 time=0.065 ms
64 bytes from 192.168.49.2: icmp_seq=4 ttl=64 time=0.043 ms
64 bytes from 192.168.49.2: icmp_seq=6 ttl=64 time=0.065 ms
64 bytes from 192.168.49.2: icmp_seq=7 ttl=64 time=0.066 ms
64 bytes from 192.168.49.2: icmp_seq=9 ttl=64 time=0.048 ms
```

**Comment by waitinfuture** [comment]
Hi @RexXiong @zhongqiangczq , could you please take a look at this PR?

File: charts/celeborn/values.yaml [hunk 1]
```diff
@@ -33,6 +33,9 @@ masterReplicas: 3
 # worker replicas set on demand, should less than node number
 workerReplicas: 5
 
+hostNetwork: false
+dnsPolicy: Default
```
Comment (zhongqiangczq) [inline charts/celeborn/values.yaml:37]: Default means use Host's Dns.
is right that dnspolicy 's default value is ClusterFirst ?

Comment (zwangsheng) [reply charts/celeborn/values.yaml]: Thanks for catching this, fixed.

File: charts/celeborn/templates/master-statefulset.yaml [hunk 2]
```diff
@@ -71,6 +71,10 @@ spec:
       securityContext:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.hostNetwork }}
+      hostNetwork: true
+      {{- end }}
+      dnsPolicy: {{ .Values.dnsPolicy | default "Default" }}
```
Comment (zhongqiangczq) [inline charts/celeborn/templates/master-statefulset.yaml:77]: default "Default" -> default "ClusterFirst"

File: charts/celeborn/templates/worker-statefulset.yaml [hunk 3]
```diff
@@ -71,6 +71,10 @@ spec:
       securityContext:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.hostNetwork }}
+      hostNetwork: true
+      {{- end }}
+      dnsPolicy: {{ .Values.dnsPolicy | default "Default" }}
```
Comment (zhongqiangczq) [inline charts/celeborn/templates/worker-statefulset.yaml:77]: default "Default" -> default "ClusterFirst"

Comment (zwangsheng) [reply charts/celeborn/templates/worker-statefulset.yaml]: Fixed Both XD

**Review by zhongqiangczq — APPROVED** [review]
LGTM

**Comment by FMX** [comment]
I think docs can be updated for this PR because there might be a scenario that some users want to use Celeborn on Spark but run workloads outside the cluster.

File: charts/celeborn/templates/master-statefulset.yaml [hunk 4]
```diff
@@ -71,6 +71,10 @@ spec:
       securityContext:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.hostNetwork }}
+      hostNetwork: true
```
[the remark this answers is not in the document]
Comment (zwangsheng) [reply charts/celeborn/templates/master-statefulset.yaml:75]: It seems to be a logical problem here, and what you're trying to express is why don't assign a value directly to hostNetwork here(correct me if I'm wrong).

I'm fine with changed to 
```yaml
hostNetwork: {{ .Values.hostNetwork | default true }}
```

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main/0.3

**Comment by RexXiong** [comment]
> @waitinfuture is adding node port svc per worker pod is something that we can considered as alternative to hostnet? (and LB svc or node port per master), I can open PR if it make sense, in our case we stable DNS name for masters and workers that would be accessed from outside the cluster, hostnet make it harder to configure

Thanks @shlomitubul, Some users have encountered issues accessing the Celeborn worker/master across Kubernetes clusters. I believe this solution should work, am I right?

**Comment by shlomitubul** [comment]
> > @waitinfuture is adding node port svc per worker pod is something that we can considered as alternative to hostnet? (and LB svc or node port per master), I can open PR if it make sense, in our case we stable DNS name for masters and workers that would be accessed from outside the cluster, hostnet make it harder to configure
> 
> Thanks @shlomitubul, Some users have encountered issues accessing the Celeborn worker/master across Kubernetes clusters. I believe this solution should work, am I right?

yes, its a common pattern in application deploy in k8s to expose nodeport / LoadBalancer service per replica, ill open PR soon

**Pull request closed without merging**